### PR TITLE
aggregate setup & req. & config in pyproject

### PR DESCRIPTION
**Pull request by Borda**
this is standard Python project configuration and as an advantage it allows us to add optional settings for pytest/back/etc. without adding a new file

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -117,7 +117,7 @@ MAX_JOBS=4 pip install 'flash-attn>=2.0.0.post1' --no-build-isolation
 All good, now install the dependencies plus some optional ones:
 
 ```bash
-pip install -r requirements.txt tokenizers sentencepiece
+pip install . tokenizers sentencepiece
```
Comment (carmocca) [inline README.md:120]: Using the project without installing the package is the intended way to use it

File: pyproject.toml
Comment (carmocca) [inline pyproject.toml:1]: It's not clear to me that this refactor can be made. It would require confirmation

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+classifiers = [
+    "Topic :: Text Processing"
+]
+
+[project.optional-dependencies]
+all = [
+    "torch>=2.1",  # note this have not been released yet
+    "jsonargparse[signatures]",  # CLI
+]
+dev = [
```
Comment (aniketmaurya) [inline pyproject.toml:27]: it also requires [lm-evaluation-harness](https://github.com/Lightning-AI/lit-gpt/blob/8f04791d6ba9ce28eef4fe3af6b9672e4445ab67/requirements.txt#L12) for evaluation

Comment (carmocca) [reply pyproject.toml:27]: There are no eval tests, so CI doesn't require it

File: pyproject.toml
Comment (Borda) [reply pyproject.toml:1]: so we can set some formatting/testing like `black` or `pytest` configs to their project file so as not to be hidden in some GH files and can be easier reproduced locally

**Pull request closed without merging**